spark/dl/src/main/scala/com/intel/analytics/bigdl/python/api/PythonBigDLOnnx.scala

```diff
@@ -16,24 +16,297 @@
+      padW, padH, nGroup, propagateBack, wRegularizer, bRegularizer,
+      initWeight, initBias, initGradWeight, initGradBias, withBias, format
+    )
+
+  }
+
+  def createGather(axis: Int): nn.ops.Gather[T, T] = {
+    OnnxOpsMapper.gather.apply()
+  }
+
+
+  def createGemm(alpha: Float, beta: Float, transA: Int, transB: Int,
+                 matrixB: JTensor, matrixC: JTensor): Gemm[T] = {
+    OnnxOpsMapper.gemm.apply(alpha, beta,
+      (if (transA == 0) false else true),
+      (if (transB == 0) false else true),
+      toTensor(matrixB), toTensor(matrixC))
+  }
+
+
+  def createMaxPool(kernelShape: JList[Int], autoPad: String,
+    ceilMode: Int, dilations: JList[Int], pads: JList[Int],
+    storageOrder: Int, strides: JList[Int]): nn.SpatialMaxPooling[T] = {
+
+    val (kW: Int, kH: Int) = kernelShape.asScala.toList match {
+      case List(width, height) => (width, height)
+      case _ => throw new IllegalArgumentException(
+        "Kernel shape is expected in the form of List(width, height)," +
+          "the input kernel shape: " + kernelShape)
+    }
+    val (dW: Int, dH: Int) = strides.asScala.toList match {
+      case null => (1, 1)
+      case List(width, height) => (width, height)
+      case _ => throw new IllegalArgumentException(
+        "Strides is expected in the form of List(width, height)," +
+          "the input strides: " + strides)
+    }
+    val (padW: Int, padH: Int) = pads.asScala.toList match {
+      case null => (0, 0)
+      case List(width, height) => (width, height)
+      case _ => throw new IllegalArgumentException(
+        "Pads is expected in the form of List(width, height)," +
+          "the input pads: " + pads)
+    }
+
+    if (ceilMode != 0) {
+      throw new IllegalArgumentException("MaxPool doesnt support ceil mode yet.")
+    }
+
+    if (storageOrder != 0) {
+      throw new IllegalArgumentException("MaxPool doesnt support storage order yet.")
+    }
+
+    val format = DataFormat.NCHW
+
+    OnnxOpsMapper.maxPool.apply(kW, kH, dW, dH, padW, padH, format)
+
+  }
+
+  def createRelu(): nn.ReLU[T] = {
+    OnnxOpsMapper.relu.apply(false)
+  }
+
+  def createReshape(shape: JArrayList[Int]): nn.Reshape[T] = {
+    OnnxOpsMapper.reshape.apply(shape.asScala.toArray, Some(false))
+  }
+
   def createShape(): Shape[T] = {
-    Shape()
+    OnnxOpsMapper.shape.apply()
   }
 
+  def createSoftmax(axis: Int = 1): nn.SoftMax[T] = {
+    OnnxOpsMapper.softmax.apply()
+  }
+
+  def createOnnxSum(inplace: Boolean = false): nn.CAddTable[T, T] = {
+    OnnxOpsMapper.sum.apply(inplace)
+  }
+
+  def createUnsqueeze(axes: JList[Int], numInputDims: Int): nn.Unsqueeze[T] = {
+    val pos = axes.asScala.toList match {
+      case List(elem) => elem + 1 // Todo
+      case _ => throw new IllegalArgumentException("Bad axes value: " + axes)
+    }
+    OnnxOpsMapper.unsqueeze.apply(pos, numInputDims)
+  }
 }
 
 
-object PythonBigDLOnnx {
+object OnnxOpsMapper {
```

Comment:
Why create this object?

Reply:
The mapper object you suggested before.

Reply:
no, I think all your mapping is in `PythonBigDLOnnx`.  `OnnxOpsMapper ` is meaningless.